frappe/model/base_document.py

```diff
@@ -298,22 +299,25 @@ def get_valid_dict(
 	) -> dict:
 		d = _dict()
 		for fieldname in self.meta.get_valid_columns():
+			field_value = getattr(self, fieldname, _DOC_DELETED_ATTR)
+
 			# column is valid, we can use getattr
-			d[fieldname] = getattr(self, fieldname, None)
+			d[fieldname] = field_value
```

Comment:
if `field_value` is `_DOC_DELETED_ATTR` and field isn't virtual, we could just `continue` here instead of setting `d[fieldname]` only to delete it later?

Also important to note that if `sanitize` is False we're continuing below which means `_DOC_DELETED_ATTR` will become part of the dict and possibly make it non-serializable.

Reply:
Removed `_DOC_DELETED_ATTR` and consequently the logic to remove unpermitted field keys from as_dict